pkg/operator/ceph/object/rgw.go

```diff
@@ -148,6 +143,24 @@ func (c *clusterConfig) startRGWPods(update bool) error {
 		}
 
 		// Generate the mime.types file after the rep. controller as well for the same reason as keyring
+		if createErr != nil && errors.IsAlreadyExists(createErr) {
+			// Always invoke ceph version before an upgrade so we are sure to be up-to-date
+			daemon := "rgw"
+			currentCephVersion, err := client.LeastUptodateDaemonVersion(c.context, c.clusterInfo.Name, daemon)
+			if err != nil {
+				logger.Warningf("failed to retrieve current ceph %s version. %+v", daemon, err)
+				logger.Debug("could not detect ceph version during update, this is likely an initial bootstrap, proceeding with c.clusterInfo.CephVersion")
+				if err := updateDeploymentAndWait(c.context, deployment, c.store.Namespace, c.clusterInfo.Name, c.clusterInfo.CephVersion); err != nil {
```

Comment:
Is the only difference between lines 153 and 158 the logging? Could we avoid the duplication with `updateDeploymentAndWait()` if we moved it below after line 161?

Reply:
Made a typo, we need `currentCephVersion`.